streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -662,37 +650,26 @@ void suspend(final boolean clean,
             }
 
             if (eosEnabled) {
-                maybeAbortTransactionAndCloseRecordCollector(isZombie);
+                // Ignore any exceptions whilee closing the record collector, i.e task producer.
+                closeRecordCollector();
             }
         }
     }
 
-    private void maybeAbortTransactionAndCloseRecordCollector(final boolean isZombie) {
-        if (!isZombie) {
```

Comment:
I think there are some callers of `suspend` that would set `clean` to false while `isZombie` to false as well, e.g. `suspendRunningTasks` in this case should we still call abortTxn?

Reply:
That call is inside the catch block of `task.suspend`. For simplicity, I think calling abortTxn is not very tempting at this point as well.

Reply:
After another thought, I'm pretty sure the purpose of this PR is not to maintain the caller of abortTxn in either mode to simplify our error handling. The other txn mechanism like InitPid will make sure to cleanup the pending transactions or through the txn timeout. However the risk of calling abortTxn during close is much higher than a normal processing loop, at least for today's producer. If the producer is in `FATAL_ERROR`, it becomes a bomb for any caller touching on its APIs except `close()`.

We could think of doing some `safe abort` operations for producer internally instead of externally, which could be done by either getting a new API or just change the default behavior of `producer.close` to do the transaction abortion when time allowed.

Reply:
Cool I think I am convinced, let's just ignore the isZombie flag then.